### PR TITLE
Ladybird: Enable and document build and debug steps with Xcode

### PR DESCRIPTION
**Pull request by ADKaster**
Tested on an M2 MacBook Air with Xcode 14.2

### CODE REVIEW

File: Documentation/Xcode_ladybird_Scheme.png
Comment (ADKaster) [inline Documentation/Xcode_ladybird_Scheme.png:1]: fwiw this file was run through `optipng` with the default settings. not sure if we've been using any extra flags on them for other files.

File: Meta/CMake/lagom_compile_options.cmake [hunk 1]
```diff
@@ -3,6 +3,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/common_compile_options.cmake)
 add_compile_options(-Wno-implicit-const-int-float-conversion)
 add_compile_options(-Wno-literal-suffix)
 add_compile_options(-Wno-maybe-uninitialized)
+add_compile_options(-Wno-shorten-64-to-32)
```
Comment (linusg) [inline Meta/CMake/lagom_compile_options.cmake:6]: This very much belongs into a separate commit

File: Documentation/LadybirdBuildInstructions.md [hunk 2]
```diff
@@ -119,6 +119,30 @@ export SERENITY_SOURCE_DIR=$(realpath ../)
 ./Build/ladybird/ladybird # or, in macOS: open ./Build/ladybird/ladybird.app
 ```
 
+### Debugging with Xcode on macOS
+
+The ``serenity.sh`` build script does not know how to generate Xcode projects, so creating the project must be done manually.
```
Comment (linusg) [inline Documentation/LadybirdBuildInstructions.md:124]: Single backticks in markdown is the style we usually use

File: Documentation/Xcode_ladybird_Scheme.png
Comment (linusg) [reply Documentation/Xcode_ladybird_Scheme.png:1]: yep, that's fine

**Review by linusg — CHANGES_REQUESTED** [review]
looks legit, some small requests

File: Meta/CMake/lagom_compile_options.cmake (hunk 1 above)
Comment (ADKaster) [reply Meta/CMake/lagom_compile_options.cmake:6]: Fair enough, moved to a separate commit just about the warning.

File: Documentation/LadybirdBuildInstructions.md (hunk 2 above)
Comment (ADKaster) [reply Documentation/LadybirdBuildInstructions.md]: Using single backticks for these new items now.
